src/main/java/com/exasol/exasoltestsetup/standalone/StandaloneExasolTestSetup.java

```diff
@@ -0,0 +1,166 @@
+package com.exasol.exasoltestsetup.standalone;
+
+import static com.exasol.exasoltestsetup.PasswordGenerator.generatePassword;
+import static com.exasol.exasoltestsetup.standalone.WaitHelper.waitUntil;
+
+import java.sql.*;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.exasol.bucketfs.Bucket;
+import com.exasol.bucketfs.SyncAwareBucket;
+import com.exasol.dbcleaner.ExasolDatabaseCleaner;
+import com.exasol.errorreporting.ExaError;
+import com.exasol.exasoltestsetup.*;
+import com.jcraft.jsch.*;
+
+/**
+ * This class implements the {@link ExasolTestSetup} interface for Exasol databases that are running in the Cloud or in
+ * a VM.
+ */
+public class StandaloneExasolTestSetup implements ExasolTestSetup {
+    private static final java.util.logging.Logger LOGGER = Logger.getLogger(StandaloneExasolTestSetup.class.getName());
+    private static final int BUCKET_FS_PORT = 2580;
+    private static final int DATABASE_PORT = 8563;
+    private final ConnectionDetailProvider connectionDetails;
+    private final String bucketFsReadPassword;
+    private final String bucketFsWritePassword;
+    private final SshConnection sshConnection;
+    private final int localBucketFsPort;
+    private final int localDatabasePort;
+    private final List<String> dataNodeIds;
+
+    /**
+     * Create a new instance of {@link StandaloneExasolTestSetup}.
+     * 
+     * @param connectionDetailProvider provider for the connection details
+     */
+    public StandaloneExasolTestSetup(final ConnectionDetailProvider connectionDetailProvider) {
+        this.connectionDetails = connectionDetailProvider;
+        this.sshConnection = setupSshConnection();
+        this.dataNodeIds = fetchDataNodeIds();
+        this.localBucketFsPort = this.sshConnection.addForwardPortForwarding(BUCKET_FS_PORT, getADataNode());
+        final int localHttpsPort = this.sshConnection.addForwardPortForwarding(443);
+        this.localDatabasePort = this.sshConnection.addForwardPortForwarding(DATABASE_PORT, getADataNode());
+        final ExaOperationInterface exaOperationInterface = new ExaOperationInterface("localhost:" + localHttpsPort,
+                connectionDetailProvider.getAdminCredentials());
+        exaOperationInterface.startStorageServiceIfNotRunning();
+        exaOperationInterface.startAllDatabases();
+        exaOperationInterface.setBucketFsPort("bfsdefault", BUCKET_FS_PORT);
+        this.bucketFsReadPassword = generatePassword();
+        this.bucketFsWritePassword = generatePassword();
+        exaOperationInterface.setBucketPasswords(this.bucketFsReadPassword, this.bucketFsWritePassword);
+        waitForExasolToSyncTheSettings();
+        waitUntil(this::isSqlInterfaceAvailable);
+        cleanTheDatabase();
```

Comment:
I wonder if we could extract this to a factory

Reply:
does not fit right now. I think I'll take a look somewhen later